obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    # new_event.depth_ind = topline[44]
+    # new_event.loc_ind = topline[45]
+    new_event.creation_info = CreationInfo(agency_id=topline[45:48].
+                                           strip())
+    ksta = Comment(text='Number of stations=' +
+                   topline[49:51].strip())
+    new_event.origins[0].comments.append(ksta)
+    # new_event.origins[0].nsta??? = _int_conv(topline[49:51])
+    if not _float_conv(topline[51:55]) == 999:
+        new_event.origins[0].time_errors['Time_Residual_RMS'] = \
+            _float_conv(topline[51:55])
+    # Read in magnitudes if they are there.
+    if len(topline[59].strip()) > 0:
+        new_event.magnitudes.append(Magnitude())
+        new_event.magnitudes[0].mag = _float_conv(topline[56:59])
+        new_event.magnitudes[0].magnitude_type = topline[59]
+        new_event.magnitudes[0].creation_info = \
+            CreationInfo(agency_id=topline[60:63].strip())
+        new_event.magnitudes[0].origin_id = new_event.origins[0].\
+            resource_id
+    if len(topline[67].strip()) > 0:
+        new_event.magnitudes.append(Magnitude())
+        new_event.magnitudes[1].mag = _float_conv(topline[64:67])
+        new_event.magnitudes[1].magnitude_type = topline[67]
+        new_event.magnitudes[1].creation_info = \
+            CreationInfo(agency_id=topline[68:71].strip())
+        new_event.magnitudes[1].origin_id = new_event.origins[0].\
+            resource_id
+    if len(topline[75].strip()) > 0:
+        new_event.magnitudes.append(Magnitude())
+        new_event.magnitudes[2].mag = _float_conv(topline[72:75])
+        new_event.magnitudes[2].magnitude_type = topline[75]
+        new_event.magnitudes[2].creation_info = \
+            CreationInfo(agency_id=topline[76:79].strip())
+        new_event.magnitudes[2].origin_id = new_event.origins[0].\
+            resource_id
+    # convert the nordic notation of magnitude to more general notation
+    for _magnitude in new_event.magnitudes:
+        _magnitude.magnitude_type = _nortoevmag(_magnitude.magnitude_type)
+    # Set the useful things like preferred magnitude and preferred origin
+    new_event.preferred_origin_id = str(new_event.origins[0].resource_id)
+    if len(new_event.magnitudes) > 1:
+        try:
+            # Select moment first, then local, then
+            mag_filter = ['MW', 'Mw', 'ML', 'Ml', 'MB', 'Mb',
+                          'MS', 'Ms', 'Mc', 'MC']
+            _magnitudes = [(m.magnitude_type, m.resource_id)
+                           for m in new_event.magnitudes]
+            preferred_magnitude = sorted(_magnitudes,
+                                         key=lambda x: mag_filter.index(x[0]))
+            new_event.preferred_magnitude_id = str(preferred_magnitude[0][1])
+        except ValueError:
+            # If there is a magnitude not specified in filter
+            new_event.preferred_magnitude_id =\
+                str(new_event.magnitudes[0].resource_id)
```

Comment:
It might be best not to set it if it is happily valid not to - there is no preferred magnitude in the nordic files.